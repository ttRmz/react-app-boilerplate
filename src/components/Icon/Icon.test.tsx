import { fireEvent, render } from "@testing-library/react"

import { Icon, ICON_NAMES } from "."

describe("<Icon />", () => {
  it("should render an icon", () => {
    const { container } = render(<Icon name={ICON_NAMES.react} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("should pass props on root element", () => {
    const onClick = jest.fn()
    const testId = "icon-id"

    const { container, getByTestId } = render(
      <Icon
        name={ICON_NAMES.react}
        className="test-class"
        id="test-id"
        onClick={onClick}
        data-testid={testId}
      />,
    )
    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(getByTestId(testId))
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
