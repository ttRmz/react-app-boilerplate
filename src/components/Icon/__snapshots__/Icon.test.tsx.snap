// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should pass props on root element 1`] = `
<i
  class="Icon Icon-react test-class"
  data-testid="icon-id"
  id="test-id"
/>
`;

exports[`<Icon /> should render an icon 1`] = `
<i
  class="Icon Icon-react"
/>
`;
